cdisc_rules_engine/dataset_builders/variables_metadata_with_library_metadata.py

```diff
@@ -0,0 +1,78 @@
+from cdisc_rules_engine.dataset_builders.base_dataset_builder import BaseDatasetBuilder
+import pandas as pd
+from typing import List
+from cdisc_rules_engine import config
+import cdisc_rules_engine.utilities.sdtm_utilities as sdtm_utilities
+
+
+class VariablesMetadataWithLibraryMetadataDatasetBuilder(BaseDatasetBuilder):
+    def build(self):
+        """
+        Returns the variable metadata from a given file.
+        Returns a dataframe with the following columns:
+        variable_name
+        variable_order_number
+        variable_label
+        variable_size
+        variable_data_type
+        has_empty_values
+        library_variable_name,
+        library_variable_label,
+        library_variable_data_type,
+        library_variable_role,
+        library_variable_core,
+        library_variable_order_number
+        """
+        # get dataset metadata and execute the rule
+        content_variables_metadata: pd.DataFrame = (
+            self.data_service.get_variables_metadata(
+                self.dataset_path, drop_duplicates=True
+            )
+        )
+        dataset_contents = self.get_dataset_contents()
+        library_variables_metadata = self.get_variables_metadata()
+
+        data = content_variables_metadata.merge(
+            library_variables_metadata,
+            how="outer",
+            left_on="variable_name",
+            right_on="library_variable_name",
+        ).fillna("")
+
+        data["has_empty_values"] = data.apply(
```

Comment:
I think it would be better to call this `variable_has_empty_values` and add it to the data_service's get_variables_metadata so that the other variable metadata builders will have access to it.

Reply:
I think that kind of defeats the purpose of the metadata reader enhancements we made in the past. We have been trying to avoid reading the entire dataset just to get metadata, would it be possible to wait on adding it to all variable metadata datasets until we know we need it?

Reply:
After a searching, it looks like there is a pythonic solution to this using `@property` decorator:
https://stackoverflow.com/a/1598204/2418023

Reply:
I'm not seeing how that would prevent us from having to read the entire dataset. Are you saying to add a property decorator to the dataset builder itself?

Reply:
Sorry, I was confusing the `data` dataframe with an object.
Maybe there's a way we could delay the computation of values within a dataframe column until it is accessed? But if it's not straightforward, then I guess it's best to keep this part in this builder as you've already done.

Reply:
Since each instance of a dataset builder is targeted to a specific domain, the variable_has_no_data column can be a property on all dataset builders and added to all variable metadata datasets. Does that address your concern? I also think that we will want to abstract all use of dataframes into their own classes anyways so it may make more sense to do it when we perform that abstraction.

Reply:
I think so, and I think it makes sense to do it with the abstraction

Reply:
I ended up not moving the "variable_has_empty_values" to the base_dataset_builder because, at least in the case it is being used for here, we need to perform the check after merging with the library metadata. This is because if a variable appears in the library metadata but not in the dataset we want to the value for "variable_has_empty_values" to be true. I believe this means it is better to calculate that column on the fly, depending on the dataset being evaluated. I realize we could just move the entire function to the base dataset builder, but I figured it would be better to do that if/when we need variable_has_empty_values in other datasets instead of making that change now.